TrafficSimulator/Assets/Car/AutoDrive.cs

```diff
@@ -124,21 +149,24 @@ void Q_SteerTowardsTarget()
             _vehicleController.steerInput = Vector3.MoveTowards(new Vector3(_vehicleController.steerInput, 0, 0), new Vector3(steeringAngle, 0, 0), Time.deltaTime).x;
         }
         void Q_UpdateTarget()
-        {            
+        {   
             // Calculate the direction, which is the vector from our current position to the target
             Vector3 direction = Q_GetTarget() - transform.position;
 
             // Calculate the dot product between our forward vector and the direction. If the target is in front of us, the dot product will be positive. If it's behind us, it will be negative
             float dot = Vector3.Dot(transform.forward, direction.normalized);
 
+            // Set new taget look ahead distance based on the current speed
+            _targetLookaheadDistance = _baseTLD * Math.Max(_vehicleController.speed / 20, 1);
+
             // If the vehicle is driving and the target is behind us and too far away
             if (_status == Status.Driving && dot < 0 && direction.magnitude > _targetLookaheadDistance + 1f)
             {
                 Debug.Log("Repositioning started, slowing down...");
                 _status = Status.RepositioningInitiated;
 
                 // Reposition to the point prior to the one we missed
-                _repositioningTargetIndex = (_targetIndex - _repositioningOffset) % _lane.Count;
+                _repositioningTargetIndex = (_targetIndex - _repositioningOffset + _lane.Count) % _lane.Count;
```

Comment:
This logic should be moved outside, for example to a `getNextIndex(int index)` function since this logic is duplicated a lot